File: frontend/musicbox/src/app/track.ts
import {Artist} from './artist';

export interface Track {
  id: string,
  name: string,
  albumId: string,
  duration_ms: number,
  artists: Artist[],
  audio: string,
}

export interface PlaylistTrack {
  track: Track,
  added_at: string,
}

File: backend/api/serializers/track.py
from rest_framework import serializers

from api.models.track import Track
from api.serializers.artist import ArtistSerializer


class TrackSerializer(serializers.ModelSerializer):
    id = serializers.ReadOnlyField()
    duration_ms = serializers.ReadOnlyField()
    artists = ArtistSerializer(many=True)

    class Meta:
        model = Track
        fields = ["id", "name", "duration_ms", "album", "artists", "audio"]
